### PR TITLE
chore: replace references to csfle shared library MONGOSH-1237

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: src/connect.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ async function connectWithFailFast(uri: string, client: MongoClient, logger: Con
     client.removeListener('serverHeartbeatFailed', heartbeatFailureListener);
     client.removeListener('serverHeartbeatSucceeded', heartbeatSucceededListener);
     logger.emit('devtools-connect:connect-attempt-finished', {
-      csfleVersionInfo: (client?.autoEncrypter as any /* NODE-4085 */)?.csfleVersionInfo
+      cryptSharedLibVersionInfo: (client?.autoEncrypter as any /* NODE-4085 */)?.cryptSharedLibVersionInfo
```
Comment (Anemy) [inline src/connect.ts]: The mentioned https://jira.mongodb.org/browse/NODE-4085 is closed - anything to do on that comment? Looks like the autoEncrypter is `js` though so we'll be keeping this `any` for the foreseeable future?

**Review by Anemy — APPROVED** [review]
lgtm! Not sure what the ci failure is about.

**Comment by addaleax** [comment]
> Not sure what the ci failure is about.

CI failure is the driver updated types since the last time we did something here :) It’s a good failure, means we can use proper typings there now.

File: src/connect.ts (hunk 1 above)
Comment (addaleax) [reply src/connect.ts]: > The mentioned https://jira.mongodb.org/browse/NODE-4085 is closed - anything to do on that comment? 

Yeah, bump it by about 200 tickets :stuck_out_tongue: NODE-4285 is the one for the renaming, we’ll need that until the next driver release.

> Looks like the autoEncrypter is `js` though so we'll be keeping this `any` for the foreseeable future?

The typings here come from the driver (which has separate typings for `AutoeEncrypter`), so this should go away at some point:)

**Review by Anemy — APPROVED** [review]

**Pull request merged**